pkg/controller.v2/controller_tfjob.go

```diff
@@ -114,3 +115,34 @@ func (tc *TFJobController) deletePodsAndServices(tfJob *tfv1alpha2.TFJob, pods [
 	}
 	return nil
 }
+
+func (tc *TFJobController) cleanupTFJob(tfJob *tfv1alpha2.TFJob) error {
```

Comment:
Can you add a unittest please?

An E2E test would also be nice. For an E2E test please file an issue if you don't do it in this PR.

Reply:
@756

Reply:
How about adding a unittest?

Reply:
Resolved